### ISSUE TITLE
Bug, exception on weapon-player collision, weapon parent is debris

### ISSUE THREAD

**Issue by EatThePath** [scp-fs2open/fs2open.github.com#3359]
An access violation in ai_ship_hit got tripped while I was testing something else in a highly chaotic mission(warmachine dev version of Universal Truth). I don't know what circumstances caused it, but I did some inspection in a debugger to try to figure out what the cause was.

I don't think this is enough to go on, but maybe in the future a pattern will emerge from any recurrences 

The error is at the end of this chunk
```cpp
		if ((hit_objp->type == OBJ_WEAPON) || (hit_objp->type == OBJ_BEAM)) {
			hitter_objnum = hit_objp->parent;
			Assert((hitter_objnum < MAX_OBJECTS));
			if (hitter_objnum == -1) {
				return; // Possible SSM, bail while we still can.
			}
			objp_hitter = &Objects[hitter_objnum];
		} else if (hit_objp->type == OBJ_SHIP) {
			objp_hitter = hit_objp;
		} else {
			Int3();	// Should never happen.
			return;
		}
		if ((hit_objp->type == OBJ_WEAPON) || (hit_objp->type == OBJ_BEAM)) {
			hitter_objnum = hit_objp->parent;
			Assert((hitter_objnum < MAX_OBJECTS));
			if (hitter_objnum == -1) {
				return; // Possible SSM, bail while we still can.
			}
			objp_hitter = &Objects[hitter_objnum];
		} else if (hit_objp->type == OBJ_SHIP) {
			objp_hitter = hit_objp;
		} else {
			Int3();	// Should never happen.
			return;
		}
		Assert(objp_hitter != NULL);
		hitter_aip = &Ai_info[Ships[objp_hitter->instance].ai_index];
```

In inspection, I determined the `hit_objp` is initially a submunition missile. That missile's parent, which `hit_objp` is set to, is somehow a debris object. The instance of that debris is well outside the bounds of the ships array, so trying to index into ships with it blows up.

>	fs2_open_21_1_0_x64_AVX2-FASTDBG.exe!ai_ship_hit(object * objp_ship, object * hit_objp, vec3d * hit_normal) Line 15253	C++
 	fs2_open_21_1_0_x64_AVX2-FASTDBG.exe!ship_apply_local_damage(object * ship_objp, object * other_obj, vec3d * hitpos, float damage, int quadrant, bool create_spark, int submodel_num, vec3d * hit_normal) Line 2433	C++
 	fs2_open_21_1_0_x64_AVX2-FASTDBG.exe!ship_weapon_do_hit_stuff(object * pship_obj, object * weapon_obj, vec3d * world_hitpos, vec3d * hitpos, int quadrant_num, int submodel_num, vec3d hit_dir) Line 118	C++
 	fs2_open_21_1_0_x64_AVX2-FASTDBG.exe!ship_weapon_check_collision(object * ship_objp, object * weapon_objp, float time_limit, int * next_hit) Line 464	C++
 	fs2_open_21_1_0_x64_AVX2-FASTDBG.exe!collide_ship_weapon(obj_pair * pair) Line 548	C++
 	fs2_open_21_1_0_x64_AVX2-FASTDBG.exe!`anonymous namespace'::obj_collide_pair(object * A, object * B) Line 999	C++
 	fs2_open_21_1_0_x64_AVX2-FASTDBG.exe!`anonymous namespace'::obj_find_overlap_colliders(std::vector<int,std::allocator<int>> & overlap_list_out, std::vector<int,std::allocator<int>> & list, int axis, bool collide) Line 1038	C++
 	fs2_open_21_1_0_x64_AVX2-FASTDBG.exe!obj_move_all(float frametime) Line 1594	C++
 	fs2_open_21_1_0_x64_AVX2-FASTDBG.exe!game_simulation_frame() Line 3561	C++
 	fs2_open_21_1_0_x64_AVX2-FASTDBG.exe!game_frame(bool paused) Line 3964	C++
 	fs2_open_21_1_0_x64_AVX2-FASTDBG.exe!game_do_state(int state) Line 6167	C++
 	[Inline Frame] fs2_open_21_1_0_x64_AVX2-FASTDBG.exe!gameseq_process_events() Line 386	C++
 	fs2_open_21_1_0_x64_AVX2-FASTDBG.exe!game_main(int argc, char * * argv) Line 6538	C++
 	fs2_open_21_1_0_x64_AVX2-FASTDBG.exe!SDL_main(int argc, char * * argv) Line 7627	C++
 	fs2_open_21_1_0_x64_AVX2-FASTDBG.exe!main_getcmdline(...) Line 177	C

**Comment by Baezon** [comment]
Gonna go ahead and close this for now, as I believe this was fixed by #3369/#3371

**Comment by Baezon** [comment]
Gonna go ahead and close this for now, as I believe this was fixed by #3369/#3371
